src/neo/IO/Caching/HashSetCache.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Neo.IO.Caching
+{
+    public class HashSetCache<T> : IEnumerable<T> where T : IEquatable<T>
```

Comment:
Did you try with a B-Tree index?

Reply:
Not yet, I think hashset is faster because its complexity is O(1) while B-Tree is O(logn).